src/Toolkit/Collection.php

```diff
@@ -961,49 +961,63 @@ public function sort(...$args): static
 		$fields = [];
 
 		foreach ($args as $arg) {
-			// get the index of the latest field array inside the $fields array
-			$currentField = $fields ? count($fields) - 1 : 0;
+			// get the index of the latest field array inside $fields
+			$field = array_key_last($fields);
 
-			// detect the type of argument
-			// sorting direction
-			$argLower = is_string($arg) ? strtolower($arg) : null;
+			// normalize $arg
+			$arg = match (true) {
+				is_string($arg) => strtolower($arg),
+				default         => $arg
+			};
 
-			if ($arg === SORT_ASC || $argLower === 'asc') {
-				$fields[$currentField]['direction'] = SORT_ASC;
-			} elseif ($arg === SORT_DESC || $argLower === 'desc') {
-				$fields[$currentField]['direction'] = SORT_DESC;
+			// $arg defines sorting direction
+			if (
+				$arg === 'asc'  || $arg === SORT_ASC ||
+				$arg === 'desc' || $arg === SORT_DESC
+			) {
+				$fields[$field]['direction'] = match ($arg) {
+					'asc'   => SORT_ASC,
+					'desc'  => SORT_DESC,
+					default => $arg
+				};
 
 			// other string: the field name
 			} elseif (is_string($arg) === true) {
-				$values = [];
-
-				foreach ($array as $key => $value) {
-					$value = $collection->getAttribute($value, $arg);
-
-					// make sure that we return something sortable
-					// but don't convert other scalars (especially numbers) to strings!
-					$values[$key] = is_scalar($value) === true ? $value : (string)$value;
-				}
-
-				$fields[] = ['field' => $arg, 'values' => $values];
+				$fields[] = [
+					'field'  => $arg,
+					'values' => A::map($array, function ($value) use ($collection, $arg) {
+						$value = $collection->getAttribute($value, $arg);
+
+						// make sure that we return something sortable
+						// but don't convert other scalars (especially numbers)
+						// to strings!
+						return match (true) {
+							is_scalar($value) => $value,
+							default           => (string)$value
+						};
```

Comment:
It's the same for me here. I can definitely live with the match clause. I'm really just not sure if it is worth it to refactor such short lines.